Pool_Properties_Design.md

```diff
@@ -0,0 +1,173 @@
+# Design to Define Storage Pool Properties
+
+**Status**: WIP
+
+**Version**: Alpha
+
+**Author(s)**: Leon Wang (@leonwanghui)
+
+## Summary
+
+This proposal is designed for standardizing the properties definition of storage pool. This design is also part of referenced implementation of LineOfService defined in [Swordfish](https://www.snia.org/sites/default/files/SMI/swordfish/V105/Swordfish_v1.0.5_Specification.pdf) specification (v1.0.5), which promises to providing a unified approach for the management of storage and servers in hyperscale and cloud infrastructure environments.
+
+## Motivation
+
+During these two months, we did a lot of work to design OpenSDS Profile (see [here](https://docs.google.com/document/d/1irNnz019j0XuW6SZNigs6QuYFOCC3uL44EkRYJiHyq8/edit?usp=sharing)). Besides the extensible [capabilties filter](https://github.com/opensds/opensds/pull/297) mechanism was also finished several days ago. But as for the design of implementation of StoragePool resource, it's still an issue unresolved.
+
+Now that we have come to a conclusion about Profile design, it would be much easier to design StoragePool because both of them are composed of LineOfService defined in Swordfish.
+
+### Goals
+
+* Redesign the definition of `ExtraSpec` contained in `StoragePoolSpec`.
+
+* Update all storage drivers to stay up with the changes.
+
+* Update the fake pool data declared in test package.
+
+### Non-Goals
+
+None
+
+## Proposal
+
+Description of the proposed solution.
+
+### Data model impact
+
+This proposal will update the definition of `StoragePoolSpec` with these changes below:
+
+* Implement `DataStorageLoS`, `IOConnectivityLoS` and `DataProtectionLos` struct which defined in Swordfish:
+```go
+// DataStorageLoS can be used to describe a service option covering storage
+// provisioning and availability.
+type DataStorageLoS struct {
+	// The enumeration literal specifies the time after a disaster that the
+	// client shall regain conformant service level access to the primary
+	// store, currently it only supports "OnlineActive","OnlinePassive",
+	// "Nearline" and "Offline".
+	// The expectation is that the services required to implement this
+	// capability are part of the advertising system.
+	RecoveryTimeObjective int64 `json:"recoveryTimeObjective,omitempty" yaml:"recoveryTimeObjective,omitempty"`
```

Comment:
Can you add a question about RTO in the Open Issues section at the end?  Basically ask how to use this when this is also provided through the replication profile.